### PR TITLE
BSON exported

### LINKED ISSUE #3242: Realm Web: Export BSON types from the package
## Goals

Users of (expecially the IIFE / CDN bundle) wants to use BSON types exported by the "realm-web" package.

## Expected Results

```
const _id = new Realm.BSON.ObjectId();
```

## Actual Results

> Thrown:
> TypeError: Cannot read property 'ObjectId' of undefined

## Version of Realm and Tooling

- Realm Web SDK Version: 0.8.1
- Node or React Native: Web

### PR DESCRIPTION
**Pull request by kraenhansen**
## What, How & Why?

This closes #3242 by exporting the BSON types from both Realm JS and Realm Web.

This is what the JSDocs look like:

Screenshot 2020-10-27 at 13 34 11

## ☑️ ToDos
* [x] 📝 Changelog entry
* [x] 📝 `Compatibility` label is updated or copied from previous entry
* [x] 🚦 Tests
* [x] 📝 Public documentation PR created or is not necessary
* [x] 💥 `Breaking` label has been applied or is not necessary

*If this PR adds or changes public API's:*
* [x] typescript definitions file is updated
* [x] jsdoc files updated
* [ ] Chrome debug API is updated if API is available on React Native

### CODE REVIEW

File: docs/bson.js [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ////////////////////////////////////////////////////////////////////////////
 //
-// Copyright 2020 Realm Inc.
+// Copyright 2016 Realm Inc.
```
Comment (kraenhansen) [inline docs/bson.js:3]: Hah! What a weird change. I didn't move this file 😆

**Review by kneth — APPROVED** [review]

File: packages/realm-web-integration-tests/src/bson.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+import { expect } from "chai";
+import { BSON } from "realm-web";
+
+describe("BSON", () => {
+    it("gets exported", () => {
+        expect(typeof BSON).equals("object");
+    });
+
+    it("can construct an ObjectId", () => {
```
[the remark this answers is not in the document]
Comment (kraenhansen) [reply packages/realm-web-integration-tests/src/bson.test.ts:27]: If constructing an ObjectId works, I would expect a failure to construct a Decimal128 to come from a bug in the `bson` package. In other word, I think its safe to keep it limited to this.

**Review by blagoev — COMMENTED** [review]
We are not validating if the BSON type is coming from our dependency or somewhere else. Are we going to do that or we are going to allow external BSON type usage. So two ways to handle these:
1) We allow the usage of external BSON types and its responsibility of the developer to handle any incompatibilities
2)We validate the BSON type is explicitly our own dependency and throw errors if it is not.
 
Currently we have these checks which will be valid regardless of where the BSON type is coming from. 
https://github.com/realm/realm-js/blob/v10/src/node/node_value.hpp#L101-L123

**Comment by kraenhansen** [comment]
I've rebased this on the current master and testet it locally in Chrome debugging mode ✅

**Comment by cristianoccazinsp** [comment]
Started to get `BSON: For React Native please polyfill crypto.getRandomValues, e.g. using: https://www.npmjs.com/package/react-native-get-random-values.` from either this or similar update. 

Any thoughts? We send warnings to our reporting tool so this is causing some really bad noise right now.

https://github.com/realm/realm-js/issues/3714

**Pull request merged**